test/component/test_component.py

```diff
@@ -129,3 +132,36 @@ def run(self, value: int):
 
     comp = MockComponent()
     assert comp.__canals_component__
+
+
+def test_default_component_to_dict():
+    MyComponent = factory.component_class("MyComponent")
+    comp = MyComponent()
+    res = _default_component_to_dict(comp)
+    assert res == {
+        "hash": id(comp),
+        "type": "MyComponent",
+        "init_parameters": {},
+    }
+
+
+def test_default_component_from_dict():
+    MyComponent = factory.component_class("MyComponent")
+    comp = _default_component_from_dict(MyComponent, {"type": "MyComponent"})
+    assert isinstance(comp, MyComponent)
+
+
+def test_default_component_from_dict_without_type():
+    with pytest.raises(ComponentDeserializationError, match="Missing 'type' in component serialization data"):
+        _default_component_from_dict(Mock, {})
+
```

Comment:
Do you think it's worth a test for additional unknown keys too? I'd say to add it, even if now it might be a bit meaningless it also doesn't cost us much

Reply:
I don't see the reason to be fair. 🤔 

If there are extra keys it's not a problem, we can still load the component from that dict.

Reply:
Yeah my idea is that if there are extra keys, something else may be wrong. But no problem, I also struggle to see what to test for, so let's leave it like this